Scripts/RemediationScripts/Remediate-SetSQLServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,886 @@
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+    }
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        Write-Host "No active Azure login session found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+
+   
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    if(-not($AutoRemediation))
+    {
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "Subscription Name: $($context.Subscription.Name)"
+        Write-Host "Subscription ID: $($context.Subscription.SubscriptionId)"
+        Write-Host "Account Name: $($context.Account.Id)"
+        Write-Host "Account Type: $($context.Account.Type)"
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "To Set minimal TLS version for SQL Servers in a Subscription, Contributor or higher privileges on the SQL Servers are required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4] Fetch all SQL Servers"
+
+    $sqlServerResources = @()
+    $requiredMinTLSVersion = 1.2
+
+
+    # To keep track of remediated and skipped resources
+    $logRemediatedResources = @()
+    $logSkippedResources=@()
+
+    $controlIds = "Azure_SQLDatabase_DP_Use_Secure_TLS_Version_Trial"
+
+     if($AutoRemediation)
+    {
+        if(-not (Test-Path -Path $Path))
+        {
+        Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)
+        Write-Host $([Constants]::SingleDashLine)
+        return
+        }
+        Write-Host "Fetching all SQL Servers failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No SQL Server(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }  
+        $validResources | ForEach-Object { 
+            try
+            {
+            $name = $_.ResourceId.Split('/')[8]
```

Comment:
resource name and resource group name are saved in the downloaded JSON, that can be used, instead of computing it again.

Reply:
used name & rg from json